File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    appeal_id: int | None = None
    type: str
    title: str
    body: str | None = None
    is_read: bool
    created_at: datetime


class PaginatedNotifications(BaseModel):
    items: list[NotificationOut]
    total: int
    unread: int
    page: int
    page_size: int

File: frontend/src/types/notification.ts
export interface Notification {
  id: number
  appeal_id?: number | null
  type: string
  title: string
  body?: string | null
  is_read: boolean
  created_at: string
}

export interface PaginatedNotifications {
  items: Notification[]
  total: number
  unread: number
  page: number
  page_size: number
}
